### PR TITLE
Builds For Supported Systems

### PR DESCRIPTION
**Pull request by glaba13**
The PR implements the structure and builds for supported builds for different CPU capabilities (**AVX/NO_AVX**).

The build is only made when BUILD_FOR_PUBLIC flag is on.

3 projects are built:

- **facemask-plugin_AVX**: plugin dll with AVX instructions
- **facemask-plugin_NO_AVX**: plugin dll without AVX instructions
- **facemask-plugin**: Plugin wrapper which loads dll depends on the CPU

This PR needs the Merge of Dlib PR at first:
https://github.com/stream-labs/dlib/pull/2

### CODE REVIEW

File: plugin/plugin_wrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+		blog(LOG_DEBUG, "[FaceMask] NO AVX");
+	}
+	return avx;
+}
+
+
+void load_dll() {
+	if (inited) {
+		return;
+	}
```
Comment (brankalakic) [inline plugin/plugin_wrapper.cpp:98]: inited? what does it mean? detected? initialized?

File: plugin/plugin_wrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+#define FACEMASK_NO_AVX		(L"facemask_NO_AVX.dll")
+
+using namespace dlib;
+
+typedef bool(*obs_module_load_fun)(void);
+typedef void(*obs_module_unload_fun)(void);
+typedef const char* (*obs_module_name_fun)();
+typedef const char* (*obs_module_description_fun)();
+typedef void(*obs_module_set_pointer_fun)(obs_module_t *module);
+typedef uint32_t(*obs_module_ver_fun)(void);
```
Comment (brankalakic) [inline plugin/plugin_wrapper.cpp]: Can you use obs_module_version_fcn. Also replace fun with fcn in others

Comment (glaba13) [reply plugin/plugin_wrapper.cpp]: Fixed

File: plugin/plugin_wrapper.cpp (hunk 1 above)
Comment (glaba13) [reply plugin/plugin_wrapper.cpp:98]: Fixed, Change to _loaded_

**Pull request merged**